Jellyfin.Server.Implementations/JellyfinDbProvider.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Jellyfin.Server.Implementations
+{
+    /// <summary>
+    /// Factory class for generating new <see cref="JellyfinDb"/> instances.
+    /// </summary>
+    public class JellyfinDbProvider
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="JellyfinDbProvider"/> class.
+        /// </summary>
+        /// <param name="serviceProvider">The application's service provider.</param>
+        public JellyfinDbProvider(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+            serviceProvider.GetService<JellyfinDb>().Database.Migrate();
```

Comment:
I think we should have users who want to do manual migrations (something I don't think we should recommend, as they would have to manually perform operations EF Core does behind the scenes) perform them themselves. If they just want to make backups, they can do that before they start the new version for the first time (Something we recommend anyway)

Reply:
Well, we can at least add that "automatic migration" flag and plain refuse to start if it's not set... but we should tell the user how to force-run the migrations with it.

The point is, people might not be doing backups on each update, but they would if the update makes the server to stop booting.

Reply:
Also, don't underestimate the number of people running Jellyfin on Docker with Watchtower that are getting the new version automatically the minute the new release is tagged.

Reply:
I think all of the reviewers know this, but I think it's worth mentioning that the migration doesn't modify the original data, except to rename the file, in fact, the connection to the database is flagged as read only, so it would take some catastrophic failures to cause any data loss. On top of that, I feel like requiring an argument to automatically perform migrations is just delaying the problem until the second migration, when it is performed automatically for all of the users who set and forget the argument. Overall, I think it's too much work for little to no benefit

Reply:
Ohm, right, old DB is retained... I'm fine with it then.

Reply:
> requiring an argument to automatically perform migrations is just delaying the problem until the second migration, when it is performed automatically for all of the users who set and forget the argument

The point of the feature is that there will be users who will not set that flag and want to perform the migrations manually. Like I mentioned originally, there are two benefits:

1. Users can have control over how/when their database is migrated
2. The Jellyfin application does not need to have schema change permission on the database

IMO I think the benefits outweigh the cost here, since the only cost being the code suggestion in my comment and adding a single property to the server config.

That being said, I suppose this migration is relatively safe, so I would be okay to put it in a future PR if you don't want to add it here

Reply:
Sounds good, it can be in a separate PR